pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1442,11 +1454,13 @@ public void seek(long timestamp) throws PulsarClientException {
     public CompletableFuture<Void> seekAsync(long timestamp) {
         if (getState() == State.Closing || getState() == State.Closed) {
             return FutureUtil
-                    .failedFuture(new PulsarClientException.AlreadyClosedException("Consumer was already closed"));
+                .failedFuture(new PulsarClientException.AlreadyClosedException(
+                    String.format("[%s][%s] Consumer was already closed", topicName.toString(), subscription)));
         }
 
         if (!isConnected()) {
-            return FutureUtil.failedFuture(new PulsarClientException("Not connected to broker"));
+            return FutureUtil.failedFuture(new PulsarClientException(
+                String.format("[%s][%s] Not connected to broker", topicName.toString(), subscription)));
```

Comment:
This error message doesn't tell what action that this exception is about. It only tells "client is not connected to a broker". It is not useful at all. You should include the action in the error message.

"Client is not connected to broker when seeking subscription %s of topic %s to timestamp %d".

Reply:
Please fix all occurrences that have similar issues.